llvm/include/llvm/IR/IntrinsicsEVM.td

```diff
@@ -136,23 +136,28 @@ def int_evm_basefee : Intrinsic<[llvm_i256_ty], [], [IntrNoMem, IntrWillReturn]>
 def int_evm_blobbasefee : Intrinsic<[llvm_i256_ty], [], [IntrNoMem, IntrWillReturn]>;
 
 // Logging.
-def int_evm_log0
-  : Intrinsic<[], [LLVMQualPointerType<AS.HEAP>, llvm_i256_ty], [IntrReadMem]>;
+def int_evm_log0 : Intrinsic<[], [LLVMQualPointerType<AS.HEAP>, llvm_i256_ty],
+                             [IntrReadMem, IntrHasSideEffects]>;
```

Comment:
Probably we could specify these intrinsics further by adding: `IntrArgMemOnly` and `ReadOnly<ArgIndex<0>>`.
What do you think?